entanglement_forging/utils/forging_subroutines.py

```diff
@@ -240,7 +307,11 @@ def _get_pauli_expectations_from_result(
     )
     pauli_names_temp = [p[1].to_label() for p in op_for_generating_circuits.paulis]
     for prep_idx, prep_string in enumerate(stateprep_strings):
-        if no_bs0_circuits and prep_string == "bs0":
+        suffix = prep_string[2]
+        bitstring_pair = [0, 0]
+        if len(prep_string) > 4:
+            bitstring_pair = [prep_string[3], prep_string[7]]
```

Comment:
un-hardcode this. We need the two numbers following the two bs strings, but we don't know the indices